crates/proto/src/xfer/dns_request.rs

```diff
@@ -54,6 +56,34 @@ pub struct DnsRequest {
 }
 
 impl DnsRequest {
+    #[cfg(feature = "std")]
+    pub(crate) fn from_query(mut query: Query, options: DnsRequestOptions) -> Self {
+        // build the message
```

Comment:
fwiw I'd be tempted to drop this comment while moving the code. It doesn't seem like it conveys any information beyond the code below it.